src/Service.Tests/CosmosTests/MutationTests.cs

```diff
@@ -242,6 +243,49 @@ public async Task MutationMissingRequiredPartitionKeyValueReturnError()
             Assert.AreEqual("The argument `_partitionKeyValue` is required.", response[0].GetProperty("message").ToString());
         }
 
+        /// <summary>
+        /// Mutation can be performed on the authorized fields
+        /// </summary>
+        [TestMethod]
+        public async Task CanCreateItemWithAuthorizedFields()
+        {
+            // Run mutation Add planet;
+            string id = Guid.NewGuid().ToString();
+            string mutation = $@"
+mutation {{
+    createEarth (item: {{ id: ""{id}"" }}) {{
+        id
+    }}
+}}";
+            JsonElement response = await ExecuteGraphQLRequestAsync("createEarth", mutation, variables: new());
+
+            // Validate results
+            Assert.AreEqual(id, response.GetProperty("id").GetString());
+        }
+
+        /// <summary>
+        /// Mutation performed on the unauthorized fields throws permission denied error
+        /// </summary>
+        [TestMethod]
+        public async Task CreateItemWithUnauthorizedFieldsReturnsError()
```

Comment:
Can we add a test where nested mutation is attempted but would fail if field auth is defined since this PR doesnt support that scenario yet?

Reply:
The nested mutation field auth is currently not support, and it won't fail, it will just check the first level and bypass for the nested fields.

Reply:
Is silently bypassing a good idea? Shouldnt we throw an exception if one of the fields that have policy on them is a nested field?

Reply:
- For nested query, the field auth is through ```@authorize``` to have policy on the field, which is currently honored. 
- For nested mutation, the field auth is through runtime config file on that entity, which we have no way to know, whether this entity is a nested entity or not just from this file, unless a schema file is traversed and a real-time query request is parsed. 
In my opinion, since before this PR, Cosmos doesn't support mutation field auth at all, this PR will just add support for top level and I'll another follow up PR to add the field auth for nested mutation, I would think it's okay. @mbhaskar What would you think about this? Any suggestions?

Reply:
> the field auth is through runtime config file on that entity, which we have no way to know, whether this entity is a nested entity or not just from this file, unless a schema file is traversed and a real-time query request is parsed.

I am not following why the runtime config file would need to know whether an entity is nested. Wouldn't you just need the name of the entity which is being evaluated (entity may or may not be nested) and evaluate permissions on that entity?

Reply:
@seantleonard I'm mentioning it in reply of this question "Shouldnt we throw an exception if one of the fields that have policy on them is a nested field?"  In order to throw exception only on the nested field auth, we would need a way to identify whether this field auth is being used as a nested level or top level.

Reply:
thanks for the clarification!